sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -598,9 +605,11 @@ template <rgba_channel_mask M> static void validate_rgba_write_channel_mask() {
 ///   undefined.
 ///
 template <rgba_channel_mask RGBAMask = rgba_channel_mask::ABGR, typename T,
-          int N>
-__ESIMD_API std::enable_if_t<(N == 8 || N == 16 || N == 32) && sizeof(T) == 4>
-scatter_rgba(T *p, simd<uint32_t, N> offsets,
+          int N, typename Toffset>
+__ESIMD_API std::enable_if_t<(N == 8 || N == 16 || N == 32) && sizeof(T) == 4 &&
+                             (std::is_same_v<Toffset, uint32_t> ||
```

Comment:
I'd suggest to use the same mechanism for template parameter checking in all APIs. Here `std::enable_if_t` is used, but in `lsc_gather` below - static_assert.
`static_assert` seems better choice, as gives clearer idea of the problem to the user.

Reply:
That is what I suggest - use static_assert everywhere